packages/excalidraw/scene/Scene.ts

```diff
@@ -95,6 +97,37 @@ class Scene {
     return this.sceneMapByElement.get(elementKey) || null;
   }
 
+  static getContainerElement = (
+    element:
+      | (ExcalidrawElement & {
+          containerId: ExcalidrawElement["id"] | null;
+        })
+      | null,
+  ) => {
+    if (!element) {
+      return null;
+    }
+    if (element.containerId) {
+      return Scene.getScene(element)?.getElement(element.containerId) || null;
+    }
+    return null;
+  };
+
+  static getBoundTextElement = (element: ExcalidrawElement | null) => {
+    if (!element) {
+      return null;
+    }
+    const boundTextElementId = getBoundTextElementId(element);
+    if (boundTextElementId) {
+      return (
+        (Scene.getScene(element)?.getElement(
+          boundTextElementId,
+        ) as ExcalidrawTextElementWithContainer) || null
+      );
+    }
+    return null;
+  };
```

Comment:
As a interim abstraction we can keep this, but the idea (as I have understood that you meant) is to make these into instance methods, not static methods. Static methods still rely on global state. The goal is to remove the element:scene mapping and thus global state.

So ultimately we want all those helpers that currently (in this PR) use these static methods to be pure (= not use global Scene state) and instead be passed elements Map or something and use that. And _then_ we'd add Scene instance methods that would use those pure helpers, passing down the state as needed.

Reply:
First I want to unblock the utils ESM minimal changes and for that reason editor will be using scene helpers where as utils / dependency of util methods will be using the pure methods (which were split from textElement).

As per refactoring, If we have an element map per excalidraw instance, then I think there is no need to move these utils to scene helpers anyway, we would be always passing the element map in that case and we need not have a separate copy of these utils as well (That AFAIR is the ultimate goal)

Reply:
> As per refactoring, If we have an element map per excalidraw instance, then I think there is no need to move these utils to scene helpers anyway, we would be always passing the element map in that case and we need not have a separate copy of these utils as well (That AFAIR is the ultimate goal)

Yes, we can keep just one helper that takes Map as an arg. But what I was pointing to above is that we _can_ have a `Scene.getContainerElement` wrapper, but it should (ultimately) be an instance method operating on instance state, not global state.